Sources/Theme.php

```diff
@@ -2730,11 +2749,23 @@ protected function loadVariant(): void
 		Utils::$context['theme_variant_url'] = '';
 
 		if (!empty($this->settings['theme_variants'])) {
+			// Add the default variant
+			$this->settings['theme_variants'] = array_unique(array_merge(['default'], $this->settings['theme_variants']));
+
 			// Overriding - for previews and that ilk.
 			if (!empty($_REQUEST['variant'])) {
 				$_SESSION['id_variant'] = $_REQUEST['variant'];
 			}
 
+			/**
+			 * Attempt to load a variants file for variable overriding
+			 * using data attribute (:root[data-variant="variant"])
+			 * 
+			 * This is useful when you only want a single file for
+			 * recoloring the variants.
+			 */
+			self::loadCSSFile('variants.css', ['order_pos' => 0], 'smf_variants');
+
```

Comment:
For JavaScript purposes.
E.g: If you defined —-primary-color-hue: 360; for red, you have a very basic red variant already. Thus loading a single file will allow to switch colors seemingly because otherwise individual files requires a page refresh.

Reply:
Additionally, the individual file will keep a certain level of "common ground" in these scenarios:
- Someone doesn't want to use JS/don't care.
- Someone previously used variants and is acquainted with the original way of using them.

Also, when we load a single file, it doesn't require to select the data attribute because it loads after anyways, overriding everything.

So overall:
- variants.css: You want to use a JS switcher or will replace very few CSS variables.
- index_variant.css: No JS, old school, no CSS variables or many changes.